scalyr_agent/client_auth.py

```diff
@@ -0,0 +1,78 @@
+import datetime
+import json
+import urllib
+from base64 import b64encode
+
+from urllib.parse import urlparse
+
+import requests
+
+
+class ClientAuth(object):
+    def __init__(self, configuration, headers):
+        self.configuration = configuration
+        self.headers = headers
+        if self.configuration.auth == "oauth2":
+            self.auth = OAuth2(self.configuration, self.headers)
+        elif self.configuration.auth == "bearer":
+            self.auth = BearerToken(self.configuration, self.headers)
+        else:
+            self.auth = NoAuth(self.configuration, self.headers)
+
+    def authenticate(self):
+        return self.auth.authenticate()
+
+class NoAuth(object):
+    def __init__(self, configuration, headers):
+        self.headers = headers
+
+    def authenticate(self):
+        # Add a header just so for traceability
+        self.headers["x-no-auth"]="true"
+        return True
+
+# Simple Authorization Header as Bearer Token using `api_key` configuration
+class BearerToken(object):
+    def __init__(self, configuration, headers):
+        headers.set("Authorization", "Bearer " + configuration.api_key)
+
+    def authenticate(self):
+        return True
+
+# Implement https://datatracker.ietf.org/doc/html/rfc6749#section-4.4 flow
+class OAuth2(object):
```

Comment:
Isn't there a python library for OAuth2 to avoid having an in-house implementation?

Reply:
Python2 requirement paints us into some really weird corners like having to be on older versions of code that opens up security risks. This seems like the path of least resistance. I originally used the OpenTelemetry Python SDK, but had to back out of that because of no Python2 support.